src/App.tsx

```diff
@@ -1,17 +1,119 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-
+import { TodoList } from './components/TodoList/TodoList';
 import users from './api/users';
+import todosFromServer from './api/todos';
+
+const preparedTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null,
+  };
+});
 
 const App: React.FC = () => {
+  const [todos, setTodos] = useState([...preparedTodos]);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const unicId = todos.length + 1;
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title) {
+      setHasTitleError(true);
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+    }
+
+    if (title && userId) {
+      const todo = {
+        id: unicId,
```

Comment:
`unicId` this variable get value from `prevState`,  so it's should be place in functional `setState`

Reply:
I chose using Date.now()